### PR TITLE
Test/ds 869 visual tests playwright

### PR DESCRIPTION
**Pull request by literat**
## Description

Introducing visual regression testing using Playwright.

### Additional context

- https://blog.appsignal.com/2023/08/16/pitfalls-to-avoid-in-playwright-for-nodejs.html
- https://lost-pixel.com/blog/post/playwright-visual-regression-testing
- https://www.checklyhq.com/blog/cypress-vs-selenium-vs-playwright-vs-puppeteer-speed-comparison
- https://playwright.dev/
- https://playwright.dev/docs/test-snapshots
- https://playwright.dev/docs/docker

### Issue reference

https://jira.lmc.cz/browse/DS-869

---

### Before submitting the PR, please make sure you do the following

- [ ] Read the [Contributing Guidelines](https://github.com/lmc-eu/spirit-design-system/blob/main/CONTRIBUTING.md).
- [ ] Follow the [PR Title/Commit Message Convention](https://github.com/lmc-eu/spirit-design-system/blob/main/CONTRIBUTING.md#commit-conventions).
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: tests/e2e/demo-homepages.spec.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    {
+      url: 'https://spirit-design-system-react.netlify.app/',
+      package: 'web-react',
+    },
+  ];
+
+  for (const demo of demos) {
+    test(`test demo homepage ${demo.package}`, async ({ page }) => {
+      await page.goto(demo.url);
+      await expect(page).toHaveScreenshot(`${demo.package}.png`, { fullPage: true, maxDiffPixelRatio: 0.01 });
```
Comment (literat) [inline tests/e2e/demo-homepages.spec.ts:18]: `maxDiffPixelRatio` - the visuals slightly differ on GitHub Actions on each run. I think it is due to the font rendering. Locally Docker does it as well but less often.

File: playwright.config.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      name: 'firefox',
+      use: { ...devices['Desktop Firefox'] },
+    },
+
+    {
+      name: 'webkit',
+      use: { ...devices['Desktop Safari'] },
+    },
+
+    /* Test against mobile viewports. */
```
Comment (literat) [inline playwright.config.ts:50]: Should I add more browsers?

File: bin/make/e2e.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+set -o errexit
+
+PLAYWRIGHT_VERSION=1.38.1
+UBUNTU_VERSION=jammy
+
+docker run --rm --network host -v $(pwd):/work/ -w /work/ -it mcr.microsoft.com/playwright:v$PLAYWRIGHT_VERSION-$UBUNTU_VERSION yarn test:e2e
```
Comment (literat) [inline bin/make/e2e.sh:8]: `Makefile` does not like the colons `:` :-/

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -161,6 +162,34 @@ This project uses Prettier for code formatting. You can run `make format` to for
...
+- checking formatting using [Prettier][prettier]
+- checking types using [Typescript][typescript] compiler
+- running unit test using [Jest][jest]
+
+### Unit testing
+
+You can run unit testing via `test:unit` scripts.
+For all available scripts see the package's `package.json` file.
+
+### Visual regression testing
```
Comment (crishpeen) [inline CONTRIBUTING.md:182]: As this PR only adds basic functionality and we are not actually able to use this for any testing, we should mention this here. When we figure out how we want to use it for testing, we can extend these docs.

Comment (literat) [reply CONTRIBUTING.md:182]: Ok, added another section.

**Pull request merged**